File: tests/fixtures/rust-syntax-family/host/crates/syntax-core/src/complexity.rs
pub fn route_invoice(value: u8) -> u8 {
    let mut score = value;
    if value > 0 {
        score += 1;
    }
    if value > 1 {
        score += 1;
    }
    if value > 2 {
        score += 1;
    }
    if value > 3 {
        score += 1;
    }
    if value > 4 {
        score += 1;
    }
    if value > 5 {
        score += 1;
    }
    if value > 6 {
        score += 1;
    }
    if value > 7 {
        score += 1;
    }
    if value > 8 {
        score += 1;
    }
    score
}

pub fn closure_decoy(value: u8) -> u8 {
    let deferred = |candidate: u8| {
        if candidate > 0 {
            return 1;
        }
        if candidate > 1 {
            return 2;
        }
        if candidate > 2 {
            return 3;
        }
        if candidate > 3 {
            return 4;
        }
        if candidate > 4 {
            return 5;
        }
        if candidate > 5 {
            return 6;
        }
        if candidate > 6 {
            return 7;
        }
        if candidate > 7 {
            return 8;
        }
        0
    };
    let _ = deferred;
    value
}

File: tests/fixtures/ruby-syntax-family/host/lib/syntax_family/complexity.rb
# frozen_string_literal: true

module SyntaxFamily
  module Complexity
    def self.route_invoice(value)
      score = value
      if value > 0
        score += 1
      end
      if value > 1
        score += 1
      end
      if value > 2
        score += 1
      end
      if value > 3
        score += 1
      end
      if value > 4
        score += 1
      end
      if value > 5
        score += 1
      end
      if value > 6
        score += 1
      end
      if value > 7
        score += 1
      end
      if value > 8
        score += 1
      end
      score
    end

    def self.block_decoy(value)
      deferred = lambda do |candidate|
        if candidate > 0
          1
        elsif candidate > 1
          2
        elsif candidate > 2
          3
        elsif candidate > 3
          4
        elsif candidate > 4
          5
        elsif candidate > 5
          6
        elsif candidate > 6
          7
        elsif candidate > 7
          8
        end
      end
      deferred.call(value)
    end
  end
end
